src/pint/utils.py

```diff
@@ -1265,6 +1266,132 @@ def split_swx(model, time):
+    to an existing WaveX model.
+
+    Parameters
+    ----------
+    fitter: PINT fitter object, None
+    freqs: iterable of float or astropy.quantity.Quantity
+        User inputed base frequencies
+    n_freqs: int, None
+        Number of wave frequencies to calculate using the equation: freq_n = 2 * pi * n / T_span
+        Where n is the wave number, and T_span is the total time span of the toas in the fitter object
+
+    Returns
+    -------
+    indices : list
+            Indices that have been assigned to new WaveX components
+    """
+    from pint.models.wavex import WaveX
+
+    if (freqs is None) and (n_freqs is None):
+        raise ValueError(
+            "WaveX component base frequencies are not specified. "
+            "Please input either freqs or n_freqs"
+        )
+
+    if (freqs is not None) and (n_freqs is not None):
+        raise ValueError(
+            "Both freqs and n_freqs are specified. Only one or the other should be used"
+        )
+
+    if n_freqs == 0:
+        raise ValueError("Must use a non-zero number of wave frequencies")
+    fitter.model.add_component(WaveX())
+    T_span = fitter.toas.get_mjds().max() - fitter.toas.get_mjds().min()
+    nyqist_freq = 1.0 / (2.0 * T_span)
+    if freqs is not None:
+        if isinstance(freqs, u.quantity.Quantity):
+            freqs.to(u.d**-1)
+        else:
+            freqs *= u.d**-1
+        if len(freqs) == 1:
+            fitter.model.WXFREQ_0001.quantity = freqs
+        else:
+            np.array(freqs)
+            freqs.sort()
+            if min(np.diff(freqs.value)) < nyqist_freq.value:
+                raise ValueError(
+                    "Wave frequency spacing is finer than frequency resolution of data"
+                )
+            fitter.model.WXFREQ_0001.value = freqs[0]
+            fitter.model.components["WaveX"].add_wavex_components(freqs[1:])
+
+    if n_freqs is not None:
+        if n_freqs == 1:
+            wave_freq = 2.0 * np.pi / T_span
+            fitter.model.WXFREQ_0001.value = wave_freq
+        else:
+            wave_numbers = np.arange(1, n_freqs + 1)
+            wave_freqs = 2.0 * np.pi * wave_numbers / T_span
+            fitter.model.WXFREQ_0001.value = wave_freqs[0]
+            fitter.model.components["WaveX"].add_wavex_components(wave_freqs[1:])
+    return fitter.model.components["WaveX"].get_indices()
+
+
+def _translate_wave_freqs(om, k):
+    """Use Wave model WAVEOM parameter to calculate a WaveX WXFREQ_ frequency parameter for wave number k
+
+    Parameters
+    ----------
+    om: float or astropy.quantity.Quantity
+        Base frequency of Wave model solution - parameter WAVEOM
+    k: int
+        wave number to use to calculate WaveX WXFREQ_ frequency parameter
+
+    Returns
+    -------
+    WXFREQ_ value or quantity that can be used in WaveX model"""
+
+    return (om * (k + 1)) / (2.0 * np.pi)
```

Comment:
You should make it so that this always returns a quantity.  If om is input as a float then you should explicitly assign units (and state what those are in the doc)

Reply:
Changed to always return a quantity